src/commands/moderation/prefixCommands/functions/setCommandPermissionSettings.ts

```diff
@@ -0,0 +1,122 @@
+    title: 'Prefix Commands - Set Permission Settings - No Connection',
+    description: 'Could not connect to the database. Unable to set the permission settings.',
+    color: Colors.Red,
+});
+
+const noCommandEmbed = (command: string) => makeEmbed({
+    title: 'Prefix Commands - Set Permission Settings - No Command',
+    description: `Failed to set default channel version for command ${command} as the command does not exist.`,
+    color: Colors.Red,
+});
+
+const failedEmbed = (command: string) => makeEmbed({
+    title: 'Prefix Commands - Set Permission Settings - Failed',
+    description: `Failed to set the permission settings for command ${command}.`,
+    color: Colors.Red,
+});
+
+const successEmbed = (command: string) => makeEmbed({
+    title: `Prefix Command permission settings set for command ${command}.`,
+    color: Colors.Green,
+});
+
+const modLogEmbed = (moderator: User, command: string, rolesBlocklist: boolean, channelsBlocklist: boolean, quietErrors: boolean, verboseErrors: boolean) => makeEmbed({
+    title: 'Prefix command permission set',
+    fields: [
+        {
+            name: 'Command',
+            value: command,
+        },
+        {
+            name: 'Roles Blocklist',
+            value: rolesBlocklist ? 'Enabled' : 'Disabled',
+        },
+        {
+            name: 'Channels Blocklist',
+            value: channelsBlocklist ? 'Enabled' : 'Disabled',
+        },
+        {
+            name: 'Quiet Errors',
+            value: quietErrors ? 'Enabled' : 'Disabled',
+        },
+        {
+            name: 'Verbose Errors',
+            value: verboseErrors ? 'Enabled' : 'Disabled',
+        },
+        {
+            name: 'Moderator',
+            value: `${moderator}`,
+        },
+    ],
+    color: Colors.Green,
+});
+
+const noModLogs = makeEmbed({
+    title: 'Prefix Commands - Set Permission Settings - No Mod Log',
+    description: 'I can\'t find the mod logs channel. Please check the channel still exists.',
+    color: Colors.Red,
+});
+
+export async function handleSetPrefixCommandPermissionSettings(interaction: ChatInputCommandInteraction<'cached'>) {
+    await interaction.deferReply({ ephemeral: true });
+
+    const conn = getConn();
+    if (!conn) {
+        await interaction.followUp({ embeds: [noConnEmbed], ephemeral: true });
+        return;
+    }
+
+    const command = interaction.options.getString('command')!;
+    const rolesBlocklist = interaction.options.getBoolean('roles-blocklist') || false;
+    const channelsBlocklist = interaction.options.getBoolean('channels-blocklist') || false;
+    const quietErrors = interaction.options.getBoolean('quiet-errors') || false;
+    const verboseErrors = interaction.options.getBoolean('verbose-errors') || false;
+    const moderator = interaction.user;
+
+    //Check if the mod logs channel exists
+    const modLogsChannel = interaction.guild.channels.resolve(constantsConfig.channels.MOD_LOGS) as TextChannel;
+    if (!modLogsChannel) {
```

Comment:
```suggestion
    let modLogsChannel = interaction.guild.channels.resolve(constantsConfig.channels.MOD_LOGS);
    if (!modLogsChannel || !modLogsChannel.isTextBased()) {
        modLogsChannel = null;
```
Might be safer to actually check if the channel is a text channel? Especially if we want to offer the codebase to other projects. Then, if it's not a text channel set it to null to prevent accidental attempts to send into a non-text channel.